packages/grid/x-grid/src/tests/sorting.XGrid.test.tsx

```diff
@@ -131,4 +140,40 @@ describe('<XGrid /> - Sorting', () => {
     apiRef.current.setSortModel(sortModel);
     expect(getColumnValues()).to.deep.equal(['Puma', 'Adidas', 'Nike']);
   });
+
+  describe('performance', () => {
+    beforeEach(() => {
+      clock.restore();
+    });
+
+    it('should sort 5,000 rows in less than 200 ms', async function test() {
+      // It's simpler to only run the performance test in a single controlled environment.
+      if (!/HeadlessChrome/.test(window.navigator.userAgent)) {
+        this.skip();
+        return;
+      }
+
+      const TestCasePerf = () => {
+        const data = useData(5000, 10);
+
+        return (
+          <div style={{ width: 300, height: 300 }}>
+            <XGrid columns={data.columns} rows={data.rows} />
+          </div>
+        );
+      };
+
+      render(<TestCasePerf />);
+      const header = screen
+        .getByRole('columnheader', { name: 'Currency Pair' })
+        .querySelector('.MuiDataGrid-colCellTitleContainer');
+
+      const t0 = performance.now();
+      fireEvent.click(header);
+      await waitFor(() => expect(document.querySelector('.MuiDataGrid-sortIcon')).to.not.be.null);
+      const t1 = performance.now();
+      const time = Math.round(t1 - t0);
+      expect(time).to.be.lessThan(200);
```

Comment:
It's not flying, could we increase the threshold?

<img width="989" alt="Screenshot 2021-04-14 at 12 39 39" src="https://user-images.githubusercontent.com/3165635/114697780-9b004d00-9d1e-11eb-97c2-d469878f9c31.png">

https://app.circleci.com/pipelines/github/mui-org/material-ui-x/4060/workflows/b303d403-165d-4802-9bf1-17d115e3c05f/jobs/18637

Also cc @DanailH that reported the issue as well

Reply:
Yes we can increase the limit and add an acceptable buffer.

Reply:
Done https://github.com/mui-org/material-ui-x/blame/43f710c6b6e6d43b2e3ef07bd4627bfee06a1021/packages/grid/x-grid/src/tests/sorting.XGrid.test.tsx#L226